### PR TITLE
Rewrite automatic citations script

### LINKED ISSUE #30: Handle case where user builds citations locally more gracefully
If a user builds citations locally, when they make a PR, github actions will also to build them, causing a conflict in the research-output file.

Handle this better. Perhaps don't let the action run if output has also been touched.

### PR DESCRIPTION
**Pull request by vincerubinetti**
- rewrite automatic citations script in response to https://github.com/greenelab/greenelab.com/pull/35
    - rename variables for clarity
    - in general, simplify and refactor
    - remove some util functions
    - make yaml structure and missing id fields critical errors
    - make manubot fail as critical error
    - match output to input by id rather than index
    - write warning to top of output file
- add force push option to hopefully fix and close #30
- remove RSS as social media icon. leave just the meta tag tag can be detected by browser extensions. if some RSS users complain that they need a clickable link to the RSS feed file to subscribe (maybe their RSS client/extension can't detect the meta tag?), we can always add it in again easily

### CODE REVIEW

**Comment by vincerubinetti** [comment]
@miltondp Could you weigh in on the RSS bullet point above, since you're the only person I know who uses RSS?

**Review by dongbohu — COMMENTED** [review]
Left a quick comment.

File: _data/generate-citations.py [hunk 1]
```diff
@@ -8,246 +8,275 @@
...
+# util
 ####################
 
-def log(message, color="white"):
-    """
-    log wih color
-    """
-    print(f"{palette[color]}{message}{palette['reset']}")
+# allow printing ANSI color codes
+os.system("")
```
[the remark this answers is not in the document]
Comment (vincerubinetti) [reply _data/generate-citations.py:28]: I got this line from a stack overflow answer, I believe it's only necessary for Windows. I could take it out.

[the remark this answers is not in the document]
Comment (dongbohu) [reply _data/generate-citations.py]: Thanks. Please feel free to close the PR if the script works fine after removing this line. I am approving it now.

**Review by dongbohu — APPROVED** [review]

**Comment by miltondp** [comment]
The RSS functionality works great. I just tried the demo and the Feedly extension found it, and I can easily add it to my feeds.

**Comment by vincerubinetti** [comment]
@miltondp Thanks! Do you know if any RSS clients or browser extensions ever need an actual clickable link on the site to the `feed.xml` to work?

I'm guessing modern RSS software probably is always able to get it from the hidden `<meta>` tag on the site.

**Comment by miltondp** [comment]
Yes, I'm pretty sure modern RSS software should find this automatically. But I'm not sure about the rest, I only used Feedly lately.

**Pull request merged**